src/core/libmaven/peakdetector.cpp

```diff
@@ -571,7 +571,7 @@ void PeakDetector::processSlices(vector<mzSlice*>& slices,
             return;
 
         _mavenParameters->allgroups.insert(
-            _mavenParameters->allgroups.begin(),
+            _mavenParameters->allgroups.end(),
```

Comment:
Does this show any affect on memory leak??

Reply:
Oh yes! Surprisingly this is the one that gave us 85% of the save. Also decreased peak-detection time by some amount but I did not measure it.

The 'insert at begin' thing was meant for when rank-based filtering happened in this method. But since v0.12.0, I moved it out to the meta-grouping stage and this insertion remained there. It is no longer required to do it this way and appending a vector at the end of a vector is much more efficient (both in terms of memory and CPU time).